gems/pending/postgres_ha_admin/failover_databases.rb

```diff
@@ -0,0 +1,78 @@
+require 'util/postgres_dsn_parser'
+require 'postgres_ha_admin/postgres_ha_logger'
+require 'pg'
+
+module PostgresHaAdmin
+  class FailoverDatabases
+    include PostgresHaAdmin
+
+    attr_reader :yml_file, :connection_hash, :log, :connection_hash
+
+    def initialize(config_dir, log_dir, connection_params_hash)
+      @yml_file = Pathname.new(config_dir).join(DB_YML_FILE)
+      @log = PostgresHaLogger.new(log_dir)
+      @connection_hash = connection_params_hash
+    end
+
+    def refresh_databases_list(connection = nil)
+      query_repmgr(connection)
+    end
+
+    def all_databases(connection = nil)
+      if File.exist?(yml_file)
+        begin
+          YAML.load_file(yml_file)
+        rescue IOError => err
+          log.error("#{err.class}: #{err}")
+          log.error(err.backtrace.join("\n"))
+          []
+        end
+      else
+        query_repmgr(connection)
+      end
+    end
+
+    def standby_databases(connection = nil)
+      all_databases(connection).select { |record| record[:type] == 'standby' }
+    end
+
+    def active_standby_databases(connection = nil)
+      all_databases(connection).select { |record| record[:type] == 'standby' && record[:active] == true }
+    end
+
+    private
+
+    def query_repmgr(connection)
+      if connection.nil?
+        connection = PG::Connection.open(connection_hash)
+        new_connection = true
+      end
+
+      result = []
+      if table_exists?(connection)
```

Comment:
I have removed ```ApplicationRecord``` and using ```pg``` gem instead for postrgres access, and there is no ```table_exists?(<table_name>)``` method on ```PG::Connection```.

```table_exists?(connection)``` does look confusing, I will change name to more appropriate.